server/platform/shared/filestore/s3store.go

```diff
@@ -618,18 +609,36 @@ func (b *S3FileBackend) RemoveDirectory(path string) error {
 	return nil
 }
 
+// prefixedPathFast is a variation of prefixedPath
+// where we don't check for the file path. This is for cases
+// where we know the file won't exist - like while writing a new file.
+func (b *S3FileBackend) prefixedPathFast(s string) string {
```

Comment:
While `...Fast` is true, it seems like it's not the primary reason for this method to exist. What if we call this `prefixedPath`, and call the other method `prefixedPathCheckLegacy` or some such?